### ISSUE TITLE
Test suite has dependency conflicts with other test suites

### ISSUE THREAD

**Issue by dstenger** [opengeospatial/ets-wfs20#97]
After having installed all test suites on beta environment (https://github.com/opengeospatial/teamengine-builder/blob/master/beta-releases/201801.csv), following error occurs when executing WFS 2.0 test suite:
___
Error in call to extension function {public java.lang.Object com.occamlab.te.TECore.callFunction(net.sf.saxon.expr.XPathContext,java.lang.String,java.lang.String,net.sf.saxon.om.NodeInfo) throws java.lang.Exception}: Exception in extension function net.sf.saxon.s9api.SaxonApiException: Error in call to extension function {public javax.xml.transform.Source org.opengis.cite.iso19142.TestNGController.doTestRun(org.w3c.dom.Document) throws java.lang.Exception}: Exception in extension function java.lang.NoClassDefFoundError: Could not initialize class org.geotools.metadata.iso.citation.Citations
___

This error can also be reproduced with current master of TEAM Engine Docker project and module [teamengine-ets-all](https://github.com/opengeospatial/teamengine-docker/tree/master/teamengine-ets-all).

After removing ```gt-metadata-18.0.jar``` from class path the error does not occur any more. However, this removal could affect other test suites.
Why does the existence of ```gt-metadata-18.0.jar``` break the WFS 2.0 test suite?

**Comment by ajanett** [comment]
The answer to why gt-metadata is a problem is that ets-wfs20 has a dependency on ets-gml which makes use of geotoolkit (geotk-metadata*). There appears to be a conflict between the geotools and the geotoolkit content.  [FYI: Geotools references start: "gt-" whereas geotoolkit reference start: "geotk-".]

The temporary solution to this is:
1) Determine with ets packages are bringing in geotools instead of geotoolkit, and
2) Switch the ets packages to use geotoolkit instead of geotools

Ultimately the solution needs to be addressed in more detail:
Determine which dependency should these packages really be utilizing and/or where some gaps need to be filled.
Switch all test suite packages to consistently use the same dependency.

I have currently found and am correcting this in the GeoPackage NSG ETS.  Please see:
https://github.com/opengeospatial/ets-gpkg12-nsg/issues/35

**Comment by dstenger** [comment]
Possible temporary solution: Replace geotk-metadata* with gt-metadata-18.0.
@keshav-nangare Can you please investigate if that is possible?

**Comment by keshav-nangare** [comment]
@dstenger 

The jar **gt-metadata-18.0.jar** found in the ets-gpkg12-nsg dependency zip. But we can see the latest master commit of the ets-gpkg12-nsg **`ajanett`** had updated the geotoolkit dependency so the `gt-metadata` dependency is not appearing in deps.zip  file anymore. I am testing with the teamengine-docker and building with teamengine-ets-all. I will update you whether the issue is reproducing or not.

**Comment by dstenger** [comment]
During the last update of Beta environment I still had to remove gt-metadata-19.1.jar manually.
Still, it is not clarified why this library breaks WFS 2.0 test suite.

**Comment by dstenger** [comment]
It should be checked if dependency gt-metadata* can be replaced by geotk-metadata* as proposed by ajanett.

**Comment by keshav-nangare** [comment]
@dstenger 

I got the exact root cause of this `gt-metadata` dependency issue, as it is still appearing in the build. The `ets-gpkg12-nsg` with version `0.5`, which is using the gt-metadata dependency so while you build the beta release it has this test suite with version 0.5.
I can see a few commits in the master of  `ets-gpkg12-nsg` with fix which is added by `@ajanett `. This will fix the `gt-metadata` dependency issue.
I have manually tested the both versions `0.5(it has the dependency)` and` current master(does not have dependency)`. The new release of this test suite will fix this issue.
